internal/resource/url.go

```diff
@@ -396,31 +395,43 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 		defer cancelFn()
 	}
 
-	if f.AWSSession == nil {
+	if f.AWSConfig == nil {
 		var err error
-		f.AWSSession, err = session.NewSession(&aws.Config{
-			Credentials: credentials.AnonymousCredentials,
+		cfg, err := config.LoadDefaultConfig(ctx, func(o *config.LoadOptions) error {
+			o.Credentials = aws.AnonymousCredentials{}
+			return nil
 		})
 		if err != nil {
 			return err
 		}
+		f.AWSConfig = &cfg
 	}
-	sess := f.AWSSession.Copy()
+	cfg := f.AWSConfig.Copy()
 
 	// Determine the partition and region this bucket is in
 	regionHint := "us-east-1"
 	if f.S3RegionHint != "" {
 		regionHint = f.S3RegionHint
 	}
-	region, err := s3manager.GetBucketRegion(ctx, sess, u.Host, regionHint)
+	region, err := manager.GetBucketRegion(ctx, s3.NewFromConfig(cfg), u.Host)
```

Comment:
s3 client should use regionHint as it's region

Reply:
done